### DISCUSSION TITLE
How to not horizontally scale right side of NineSlicePlane

### DISCUSSION THREAD

**Discussion by xzilja** [pixijs/pixijs#7110] [Q&A] [answered]
Code sandbox (you might need to refresh codesandbox web view if nothing shows up): https://codesandbox.io/s/pixi-nine-slice-plane-bar-jsr2c

I've been stuck with use case above. I am trying to make that mana bar graphic dynamic in a way that allows me to set its width based on current / full mana values. Issue I am facing now is with that rounded triangle corner. I can't figure out what values I need to use in NineSlicePlane to ensure that corner stays in tact / keeps its original aspect ratio, while body of mana bar freely stretches to grow / shrink it.

In addition to that, I am seeing weird behaviour when I change height of NineSlicePlane to something smaller than original graphic, It sort of "squishes" the gradient instead of scaling down properly.

Would appreciate any pointers here, probably can't figure it out due to inexperience with NineSlicePlane concept.

**Comment by bigtimebuddy** [comment, marked as answer]
![Screen Shot 2020-12-29 at 11 14 23 AM](https://user-images.githubusercontent.com/864393/103308318-0704b280-49c7-11eb-9dbc-ee182382619c.png)
Try this:

```js
new NineSlicePlane(manaBarTexture, 0, 0, 20, 0);
```
Those values are left, top, right, bottom.
